drivers/wifi/siwx917/siwx917_wifi.c

```diff
@@ -311,6 +312,107 @@ static void siwx917_ethernet_init(struct net_if *iface)
 	}
 }
 
+#ifdef CONFIG_WIFI_SIWX917_AP_MODE
+static int siwx917_ap_enable(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct siwx917_dev *sidev = dev->data;
+	sl_wifi_ap_configuration_t configuration;
+	int ret;
+
+	__ASSERT(params, "params should not be NULL");
+
+	configuration.channel.channel = params->channel;
+	configuration.encryption = SL_WIFI_CCMP_ENCRYPTION;
+	configuration.rate_protocol = SL_WIFI_RATE_PROTOCOL_AUTO;
+	configuration.options = 0;
+	configuration.credential_id = SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID;
+	configuration.keepalive_type = SL_SI91X_AP_NULL_BASED_KEEP_ALIVE;
+	configuration.beacon_interval = CONFIG_WIFI_SIWX917_BEACON_INTERVAL;
+	configuration.client_idle_timeout = 0xFF;
+	configuration.dtim_beacon_count = CONFIG_WIFI_SIWX917_DTIM_BEACON_COUNT;
+	configuration.maximum_clients = CONFIG_WIFI_SIWX917_MAX_STATIONS;
+	configuration.beacon_stop = 0;
+	configuration.tdi_flags = SL_WIFI_TDI_NONE;
+	configuration.is_11n_enabled = 1;
+	configuration.ssid.length = params->ssid_length;
+
+	if (params->band != WIFI_FREQ_BAND_2_4_GHZ) {
+		LOG_ERR("Invalid band");
+		return -EINVAL;
+	}
+
+	configuration.channel.band = SL_WIFI_BAND_2_4GHZ;
+
+	switch (params->bandwidth) {
+	case WIFI_FREQ_BANDWIDTH_20MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_20MHz;
+		break;
+	case WIFI_FREQ_BANDWIDTH_40MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_40MHz;
+		break;
+	case WIFI_FREQ_BANDWIDTH_80MHZ:
+		configuration.channel.bandwidth = SL_WIFI_BANDWIDTH_80MHz;
+		break;
+	default:
+		LOG_ERR("Invalid bandwidth");
+		return -EAGAIN;
+	}
+
+	strncpy(configuration.ssid.value, params->ssid, params->ssid_length);
+
+	switch (params->security) {
+	case WIFI_SECURITY_TYPE_NONE:
+		configuration.security = SL_WIFI_OPEN;
+		break;
+
+	case WIFI_SECURITY_TYPE_PSK:
+		configuration.security = SL_WIFI_WPA2;
+		sl_net_wifi_psk_credential_entry_t wifi_ap_credential = {
+			.type = SL_NET_WIFI_PSK, .data_length = params->psk_length};
```

Comment:
Avoid statements after declarations.

BTW, what is purpose of this struct? Why not:
```
		ret = sl_net_set_credential(SL_NET_DEFAULT_WIFI_AP_CREDENTIAL_ID,
					    SL_NET_WIFI_PSK, params->psk,
					    params->psk_length);
```